examples/AvroGeneric/Program.cs

```diff
@@ -68,23 +68,29 @@ static async Task Main(string[] args)
                 {
                     consumer.Subscribe(topicName);
 
-                    while (!cts.Token.IsCancellationRequested)
+                    try
                     {
-                        try
+                        while (true)
                         {
-                            var consumeResult = consumer.Consume(cts.Token);
+                            try
+                            {
+                                var consumeResult = consumer.Consume(cts.Token);
 
-                            Console.WriteLine($"Key: {consumeResult.Message.Key}\nValue: {consumeResult.Value}");
-                        }
-                        catch (ConsumeException e)
-                        {
-                            Console.WriteLine("Consume error: " + e.Error.Reason);
+                                Console.WriteLine($"Key: {consumeResult.Message.Key}\nValue: {consumeResult.Value}");
+                            }
+                            catch (ConsumeException e)
+                            {
+                                Console.WriteLine("Consume error: " + e.Error.Reason);
+                            }
                         }
                     }
-
-                    consumer.Close();
+                    catch (OperationCanceledException)
+                    {
+                        // commit final offsets and leave the group.
+                        consumer.Close();
+                    }
                 }
-            }, cts.Token);
```

Comment:
it doesn't hurt having this here, but it won't do anything in practice in this case so it's better to remove it or people might think it's important. what this does is tell the task scheduler to check if the token is cancelled before running the task. in this case, it never will be.